### ISSUE TITLE
Add windows support for Checker Framework

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#12695]
From the extensive discussion in https://github.com/checkstyle/checkstyle/issues/12221 and https://github.com/checkstyle/checkstyle/pull/12366

Currently, Checker Framework can't be run on Windows OS as the content of logs in Windows and Linux vary.

From Windows logs:

```
M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\Checker.java:[342,20] [argument] incompatible argument for parameter args of Violation.
  found   : @Initialized @Nullable String @Initialized @NonNull []
[WARNING] M:\checkstyleWorkspace\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\Checker.java:[342,53] [argument] incompatible argument for parameter moduleId of Violation.
  found   : null (NullType)
[WARNING] 
  required: @Initialized @NonNull String
```
The corresponding violation is:
```xml
  <checkerFrameworkError unstable="false">
    <fileName>src\main\java\com\puppycrawl\tools\checkstyle\Checker.java</fileName>
    <specifier>argument</specifier>
    <message>incompatible argument for parameter args of Violation.</message>
    <lineContent>new String[] {ioe.getMessage()}, null, getClass(), null));</lineContent>
    <details>
      found   : @Initialized @Nullable String @Initialized @NonNull []
    </details>
  </checkerFrameworkError>
```
this should have been:
```xml
  <checkerFrameworkError unstable="false">
    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/Checker.java</fileName>
    <specifier>argument</specifier>
    <message>incompatible argument for parameter args of Violation.</message>
    <lineContent>new String[] {ioe.getMessage()}, null, getClass(), null));</lineContent>
    <details>
      found   : @Initialized @Nullable String @Initialized @NonNull []
      required: @Initialized @NonNull Object @Initialized @NonNull []
    </details>
  </checkerFrameworkError>
```
**Now this isn't related to logging or the order of logs.** The line `required: @Initialized @NonNull Object @Initialized @NonNull []` isn't present anywhere in the checker-framework logs.

This needs to be reported to Checker Framework maintainers. 

After feedback/resolution from them, we need to:

- [ ] Change our approach to not depend on the order of output, if after resolution order of output varies. (Logic at https://github.com/checkstyle/checkstyle/pull/12366)
- [ ] Create CI task to run one checker profile in windows (Complete work at https://github.com/checkstyle/checkstyle/pull/12368)

Notes:

- Fix for forward v/s backward slash is merged at https://github.com/checkstyle/checkstyle/pull/12525

**Comment by rnveach** [comment]
> Fix for forward v/s backward slash is merged at 

I would remove this from this issue and correct the output if it is already fixed and merged. It will just cause noise for this issue.

> This needs to be reported to Checker Framework maintainers

Please report our issues to them and provide the links to their issues here.

====

@romani Lets approve this issue as we work on it.

**Comment by romani** [comment]
approved.
@rnveach , is it still an issues ? I remember we did bunch of changes around this.

**Comment by rnveach** [comment]
The slashes are not an issue, which is why I said to remove them from the issue. 

> The line required: @Initialized @NonNull Object @Initialized @NonNull [] isn't present anywhere in the checker-framework logs.

This is still an issue in the checker util itself. We need to report issues to them and post the link to the issues here for monitoring.

**Comment by romani** [comment]
@rnveach , is this different issues than https://github.com/checkstyle/checkstyle/issues/12221#issuecomment-1411094125 ?

**Comment by rnveach** [comment]
#12221 is supper massive issue that is just for dicussions now. This issue is suppose to be more direct, this is what is wrong and will be fixed.

**Comment by Vyom-Yadav** [comment]
Reported to Checker Framework maintainers at https://github.com/typetools/checker-framework/issues/5665
